### ISSUE TITLE
Support new neo4j v3 () syntax

### ISSUE THREAD

**Issue by vasspilka** [neo4jrb/neo4j#1187]
There is no support for new syntax in neo4j.
Two sample errors below. Note the arrows showing where in line there should be parenthesis ()

```
Neo4j::Session::CypherError: Parentheses are required to identify nodes in patterns, i.e. (from_node) 

"MATCH from_node, to_node WHERE (ID(from_node) = {ID_from_node}) AND (ID(to_node) = {ID_to_node}) CREATE UNIQUE from_node-[rel:RESPONSE {rel_create_props}]->to_node"
       ^          ^                                                                                             ^                                            ^
 
"MATCH (previous:`SomeModel`) WHERE (ID(previous) IN {ID_previous}) OPTIONAL MATCH previous-[rel1:`RESPONSE`]->(next:`SomeModel`) RETURN ID(previous), collect(next)"
                                                                                   ^

```

Queries should have been
```
MATCH (from_node), (to_node) WHERE (ID(from_node) = {ID_from_node}) AND (ID(to_node) = {ID_to_node}) CREATE UNIQUE (from_node)-[rel:RESPONSE {rel_create_props}]->(to_node)
```
and

```
MATCH (previous:`SomeModel`) WHERE (ID(previous) IN {ID_previous}) OPTIONAL MATCH (previous)-[rel1:`RESPONSE`]->(next:`SomeModel`) RETURN ID(previous), collect(next)
```

neo4j    version: `3.0`
neo4jrb version:  `7.0.1`

**Comment by cheerfulstoic** [comment]
Just made a couple of PRs to address this.  Would like to release them as part of `neo4j` 7.0.x and `neo4j-core` 6.1.x.

There are still a few tests failing with neo4j 3.0, but this should mostly take care of things

**Comment by cheerfulstoic** [comment]
Looking for a :+1: from @subvertallchris to release patch versions

**Comment by subvertallchris** [comment]
LGTM!

**Comment by cheerfulstoic** [comment]
#1211 takes care of the most recent of these that we've found.  There have been a number of other fixes for the parens in Neo4j 3.0.  Hopefully we've got them all, but if not feel free to comment again / make a new issue!
